docs/concepts/aggregation.md

```diff
@@ -0,0 +1,165 @@
+addon API servers can use its delegated RequestHeader authentication.
+
+Enabling the aggregator in 1.6 is outside the scope of this document.  The
+aggregator directory itself provides examples on how to do so in the
+Kubernetes 1.6 release.
+
+## kubectl and Discovery
+
+When `kubectl get fl` is called, `kubectl` does not initially know what an
+`fl` is, or how to get one.  In order to determine this information, it
+uses a mechanism called *discovery*.
+
+All Kubernetes API servers serve discovery information.  To get this
+information, `kubectl` first queries `https://$SERVER/apis`, which lists
+all available API groups and versions, as well as the *preferred version*.
+Then, for each of these API groups and versions, it queries
+`https://$SERVER/apis/$GROUP/$VERSION`.  This returns a list of resources,
+along with whether the resource is *namespaced*, which operations (e.g.
+`get` or `list`) it supports, as well as any short names (like `fl` for
+`fluders`).
+
+Once `kubectl` has retrieved the entire set of available resources for
+a particular cluster, it can then determine how to operate on the exposed
+resources.  For example, in our case above, `kubectl` determines that `fl`
+is a shortname for `flunders`, and that `fluders` are a namespaced
+resource that's part of the API group `wardle`, which has a preferred
+version of `v1alpha`.  Thus, `kubectl` will attempt a query of
+`https://$SERVER/apis/wardle/v1alpha1/namespaces/$NS/flunders/`.
+
+Since querying all of these discovery endpoints for every kubectl request
+would be expensive, `kubectl` caches this information in the same folder
+as where your kubeconfig goes, based on the name of the cluster to which
+you are connecting. Clearing this cache can resolve issues with not being
+able to use `kubectl` to fetch resources from newly registered APIs.
+
+Controller managers also commonly use discovery information to determine
+whether or not they should run: if the resources that they require are not
+present of the cluster, controller managers can just fail to start.
+
+You can see the list of API group versions returned by discover using the
+`kubectl api-versions` command.
+
+## Registering APIs
+
+In order for your API to appear in the discovery information served by the
+aggregator, it must be registered with the aggregator.  In order to do
+this, the aggregator exposes a normal Kubernetes API group called
+`apiregistration.k8s.io`, with a single resource, APIService.
+
+Each APIService corresponds to a single group-version, and different
+versions of a single API can be backed by different APIService objects.
+
+Let's take a look at the APIService for `wardle/v1alpha1`, using `kubectl
+get apiservice v1alpha1.wardle -o yaml`:
+
+```yaml
+apiVersion: apiregistration.k8s.io/v1beta1
+kind: APIService
+metadata:
+  name: v1alpha1.wardle
+  ...
+spec:
+  caBundle: <base64-encoded-serving-ca-certificate>
+  group: wardle
+  version: v1alpha1
+  priority: 100
```

Comment:
late breaking change made this `groupPriority` and `versionPriority` and reversed the sort order.